test/controllers/fhir/r4/api_controller_test.rb

```diff
@@ -1224,33 +1245,54 @@ def setup_patients
   end
 
   test 'should be unprocessable entity via update with validation errors' do
+    patient = IO.read(file_fixture('fhir_invalid_patient.json'))
+    json_patient = JSON.parse(patient)
     put(
       '/fhir/r4/Patient/1',
-      params: IO.read(file_fixture('fhir_invalid_patient.json')),
+      params: patient,
       headers: { 'Authorization': "Bearer #{@system_patient_token_rw.token}", 'Content-Type': 'application/fhir+json' }
     )
     assert_response :unprocessable_entity
     json_response = JSON.parse(response.body)
-    errors = json_response['issue'].map { |i| i['diagnostics'] }
-
-    msg = 'Expected validation error on '
-    assert_equal 18, errors.length
-    assert(errors.any?(/Invalid.*Monitoring Plan/), msg + 'Monitoring Plan')
-    assert(errors.any?(/Old York.*State/), msg + 'State')
-    assert(errors.any?(/0000.*Ethnicity/), msg + 'Ethnicity')
-    assert(errors.any?(/High noon.*Preferred Contact Time/), msg + 'Preferred Contact Time')
-    assert(errors.any?(/On FHIR.*Sex/), msg + 'Sex')
-    assert(errors.any?(/Dumbphone.*Telephone Type/), msg + 'Phone Type')
-    assert(errors.any?(/123.*Primary Telephone/), msg + 'Primary Telephone')
-    assert(errors.any?(/Date of Birth/), msg + 'Date of Birth')
-    assert(errors.any?(/Last Date of Exposure/), msg + 'Last Date of Exposure')
-    assert(errors.any?(/1492.*Symptom Onset/), msg + 'Symptom Onset')
-    assert(errors.any?(/1776.*Additional Planned Travel Start Date/), msg + 'Additional Planned Travel Start Date')
-    assert(errors.any?(/2020-01-32.*Date of Departure/), msg + 'Date of Departure')
-    assert(errors.any?(/9999-99-99.*Date of Arrival/), msg + 'Date of Arrival')
-    assert(errors.any?(/Last Name/), msg + 'Last Name')
-    assert(errors.any?(/10000.*Assigned User/), msg + 'Assigned User')
-    assert(errors.any?(/Email.*Primary Contact Method/), msg + 'Email')
+    issues = json_response['issue']
+
+    assert_equal 18, issues.length
+    monitoring_plan_iss = issues.find { |e| /Invalid.*Monitoring Plan/.match(e['diagnostics']) }
+    assert(FHIRPath.evaluate(monitoring_plan_iss['expression'].first, json_patient) == 'Invalid')
+    state_iss = issues.find { |e| /Old York.*State/.match(e['diagnostics']) }
+    assert(FHIRPath.evaluate(state_iss['expression'].first, json_patient) == 'Old York')
+    eth_iss = issues.find { |e| /0000.*Ethnicity/.match(e['diagnostics']) }
+    assert_not_nil eth_iss
+    # FHIRPath lib errors on paths with nested extensions. The below assertion should work.
+    # assert(FHIRPath.evaluate(eth_iss['expression'].first, json_patient) == '0000')
```

Comment:
Is there an issue that this can be linked to so we can check back and know when it's resolved so this can be uncommented? I didn't see an issue on the github page but maybe i was looking in the wrong place.

Reply:
I created an issue https://github.com/fhir-crucible/fhir_models/issues/82.

And I decided to link it in the source where it's relevant rather than here in the test. https://github.com/SaraAlert/SaraAlert/pull/743/commits/7c86dcbe6481df1a8ebebd027c6b6e9a819845b1